### DISCUSSION TITLE
Table Pagination

### DISCUSSION THREAD

**Discussion by natankeddem** [zauberzeug/nicegui#1643] [Ideas / Feature Requests]
The `ui.table` element currently has a pagination argument that only allows setting the `rowsPerPage` property in the object injected into the `pagination` prop. There are several more useful attributes which could be exposed by making this a dictionary input. This doesn't need to be a breaking change as the type of the input can be determined and the old behaviour can be retained if only integers are passed.
For reference:
https://quasar.dev/vue-components/table

There are workarounds to this issue, if no change is the determined path forward. I couldn't get the `props` method to parse the dict\object for the life of me; I iterated for a bit trying different combinations of quoting/unquoting/single/double/triple but never got something that worked. I ended up just utilizing `_props` directly which probably isn't recommended.
```py
import os
import random
from nicegui import ui


def add():
    item = os.urandom(10 // 2).hex()
    table.add_rows({'id': item, 'count': random.randint(0, 100)})


ui.button('add', on_click=add)
columns = [
    {'name': 'id', 'label': 'ID', 'field': 'id', 'sortable': True},
    {'name': 'count', 'label': 'Count', 'field': 'count'},
]
table = ui.table(columns=columns, rows=[], row_key='id').classes('w-full')
table.props('binary-state-sort')
# p = {'rowsPerPage': 0, 'sortBy': 'id', 'descending': True}
# table.props(
#     '''pagination="{"rowsPerPage": 0, "sortBy": "id", "descending": true}"''')
# table.props(f'pagination="{p}"')
table._props['pagination'] = {
    'rowsPerPage': 0, 'sortBy': 'id', 'descending': True}
table.update()

ui.run()
```

**Comment by falkoschindler** [comment]
That's a great idea, @natankeddem!

Using `_props` is fine (if you know what you're doing). But you're right, it's no official API.

Would you like to create a PR for it? If we simply allow passing the `pagination` dictionary instead of an `int`, the change should be rather small.

**Comment by natankeddem** [comment]
Yes, I'll create a PR. Just to be clear, from your statement, it sounds like you don't want to remain backwards compatible? It shouldn't be difficult to retain backwards compatibility but if you want to remove the complexity of handling `int` and `dict` types that is understandable.

**Reply by falkoschindler** [reply]
Oh, sorry for the confusion: I _do_ think we should preserve backwards compatibility.

**Comment by falkoschindler** [comment]
This feature has been implemented in PR #1646.
